verticapy/learn/metrics.py

```diff
@@ -508,29 +434,21 @@ def quantile_error(
     """
     check_types(
         [
-            ("q", q, [int, float],),
-            ("y_true", y_true, [str],),
-            ("y_score", y_score, [str],),
-            ("input_relation", input_relation, [str, vDataFrame],),
+            ("q", q, [int, float]),
+            ("y_true", y_true, [str]),
+            ("y_score", y_score, [str]),
+            ("input_relation", input_relation, [str, vDataFrame]),
         ]
     )
-    cursor, conn, input_relation = check_cursor(cursor, input_relation)
-    query = "SELECT APPROXIMATE_PERCENTILE(ABS({} - {}) USING PARAMETERS percentile = {}) FROM {} WHERE {} IS NOT NULL AND {} IS NOT NULL;".format(
-        y_true, y_score, q, input_relation, y_true, y_score,
-    )
-    executeSQL(cursor, query, "Computing the Quantile Error.")
-    result = cursor.fetchone()[0]
-    if conn:
-        conn.close()
-    return result
+    query = "SELECT APPROXIMATE_PERCENTILE(ABS({} - {}) USING PARAMETERS percentile = {}) FROM {} WHERE {} IS NOT NULL AND {} IS NOT NULL;".format(y_true, y_score, q, input_relation if isinstance(input_relation, str) else input_relation.__genSQL__(), y_true, y_score)
```

Comment:
You do this so many times that it might make sense to make create_metric_query helper function which accepts the string that changes (e.g. APPROXIMATE_PERCENTILE(ABS({} - {}) USING PARAMETERS percentile = {})) and fills in everything else. Then, if this every changes in the future you will only have to change it in one place.